### DISCUSSION TITLE
Code-server:Failed to save "filenew.cpp": Unable to write file (NoPermissions (FileSystemError): Error: EACCES: permission denied, open '/home/.....')

### DISCUSSION THREAD

**Discussion by jesusdeveloper** [coder/code-server#6047] [Q&A]
I use docker ubuntu with code-server installed and in it I have mounted with webdav a Nextcloud directory.
When I access docker the username (which has davfs2 permissions) can see Nextcloud files in code-server.
When it saves it shows the message:

> Failed to save "filenew.cpp": Unable to write file
> "vscode-remote://myip:port/home/nextcloud/asdasdas/filenew.cpp"
> (NoPermissions (FileSystemError): Error: EACCES: permission denied,
> open '/home/nextcloud/asdasdas/filenew.cpp')

The solution is easy, I run the following:

    sudo chown -R abc /home/nextcloud/asdasdas/filenew.cpp

And now the user abc can save the file and modify it.

I can use chown for whole directories and that's fine. The problem is that if another linux user creates a new file, if I want to edit it I have to run sudo chown..... again.

My question is, is there a way to give chown permissions to a user for a directory even if new files are added?

THANK YOU

**Comment by lg-nineoneninethreenine** [comment]
Perhaps, you make a user group, make sure the directory is in the group, and the files are also in said group. Then, you add your other user to the same group. Maybe then, access to files can be globalised. Hope it helps :)

**Comment by autumn-stevehuynh** [comment]
In VSC, I used an extension "yy0931.save-as-root"
but It could not be found in code-server.

**Reply by code-asher** [reply]
code-server uses Open VSX (https://open-vsx.org/) so please ask the extension author to publish it there!  https://github.com/eclipse/openvsx/wiki/Publishing-Extensions#how-to-publish-an-extension

**Comment by wanghan17** [comment]
@jesusdeveloper 
hello
when i use sudo chmod
code server need password
what is the password of abc？
thank you a lot
